pidtree_bcc/probes/__init__.py

```diff
@@ -4,6 +4,7 @@
 import re
 from datetime import datetime
 from multiprocessing import SimpleQueue
+from threading import Thread
```

Comment:
This makes me feel a little itchy because of the GIL despite the fact that I don't have any "better" suggestion. Whilst it strikes me that we could equally do this checkpointing in a separate, dedicated process using the same codebase (therefore with access to exactly the same process enrichment), I think the serialization of the output might cause problems and there's also no sense in "prematurely" optimizing this case because it's going to be lower volume than the tcp connect probe anyway, so I guess the "batteries included" approach here makes most sense.

Reply:
I went for threads because this operation is effectively pretty quick, IO-bound and should not be executed too often, so the GIL penalty becomes not very impactful. I'd probably go for keeping this implicit performance restriction in place as these "sidecars" should ideally not ever do any real heavy lifting.